src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -782,4 +794,197 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
             "details": errors_by_operation,
             "message": message
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
     # endregion
+
+    # region - Patch Truncation
+    def log_truncated_removed_packages_if_any(self):
+        if not len(self.__assessment_truncated_removed) == 0:
+            self.composite_logger.log_debug("Assessment truncated removed packages : {0}".format(self.__assessment_truncated_removed))
+            return True
+        return False
+
+    def __write_truncated_status_file(self, status_file_payload):
+        """ Truncate the substatus summary patch list when complete status file size is more than 126kb """
+        complete_status_byte_size = self.__get_byte_size(status_file_payload)
+        truncated_status_file = status_file_payload
+        tombstone_truncated_map = {}
+
+        if complete_status_byte_size > Constants.StatusTruncationConfig.AGENT_STATUS_FILE_SIZE_LIMIT_IN_BYTES:
+            self.composite_logger.log_debug("Begin Truncation")
+            errors_detail_list = []
+
+            # Truncated assessment patch when operation is not installation
+            if self.execution_config.operation != Constants.INSTALLATION and len(self.__assessment_packages) > 0:
+                # Perform assessment truncation
+                assessment_truncated_packages, truncated_packages_removed = self.__assessment_truncation_helper_func(self.__assessment_packages, Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES)
```

Comment:
I'm finding truncated_packages_removed to be a little confusing. How about packages_removed_from_assessment

So the 2 variables here would be packages_truncated_in_assessment, packages_removed_from_assessment.